Website/src/app/components/map.tsx

```diff
@@ -239,12 +280,28 @@ function Map({
 
 	// Schedule various effects (JS run after the page is rendered) for changes to various state variables.
 	useEffect(insertMap, []);
-	useEffect(setMapZoom, [initial_zoom_level]);
+	useEffect(addMapEvents, [setPosition, setZoomLevel]);
+	useEffect(setMapZoom, [zoomLevel]);
 	useEffect(setMapPosition, [position]);
 	useEffect(addTrackPath, [track_data?.path, track_data]);
 	useEffect(updateMarkers, [focus, setFocus, vehicles]);
 	useEffect(addPOIs, [points_of_interest, enriched_poi_types]);
 
+	// set the width and height of all poi icons using an effect to prevent re-rendering the icons
+	useEffect(() => {
+		// Iterate over all poi icons currently present
+		for (const poiIcon of document.querySelectorAll(".poi-icon")) {
+			if (poiIcon instanceof HTMLElement) {
+				// set the height and width using inline styles.
+				// this will probably make this component much more fragile than it needs to be...
+				poiIcon.style.width = poiIcon.style.height = `${poiIconSideLength}rem`;
```

Comment:
React does not recommend to dynamically inject a style-tag as this could impact performance. Therefore this is using inline-styles.